src/components/network-map-tab.js

```diff
@@ -804,31 +809,50 @@ export const NetworkMapTab = ({
         if (refIsMapManualRefreshEnabled.current && isInitialized) {
             return;
         }
+        // as long as rootNodeId is not set, we don't fetch any geodata
+        if (!rootNodeId) {
+            return;
+        }
         // Hack to avoid reload Geo Data when switching display mode to TREE then back to MAP or HYBRID
         // TODO REMOVE LATER
         if (!reloadMapNeeded) {
             return;
         }
-        if (!isInitialized) {
-            loadMapEquipments();
-            loadAllGeoData();
+        if (!isRootNodeDataLoaded) {
+            loadDefaultNodeMapEquipments();
+            loadRootNodeGeoData();
         } else {
             updateMapEquipmentsAndGeoData();
         }
-        setInitialized(true);
         // Note: studyUuid and dispatch don't change
     }, [
+        rootNodeId,
         disabled,
         studyUuid,
         currentNode,
-        loadMapEquipments,
+        loadDefaultNodeMapEquipments,
         updateMapEquipmentsAndGeoData,
-        loadAllGeoData,
+        loadRootNodeGeoData,
+        isRootNodeDataLoaded,
         isInitialized,
         reloadMapNeeded,
         updatedSubstationsIds,
     ]);
 
+    useEffect(() => {
```

Comment:
Maybe we could avoid this useEffect by chaining loadRootNodeGeoData and loadMissingGeoData ? 
But we need the mapEquipments to be loaded to loadMissingGeoData...

Reply:
Ok to keep this hook